programs/svm-spoke/src/instructions/bundle.rs

```diff
@@ -129,62 +133,24 @@ where
         relayer_refund_leaf.leaf_id,
     );
 
-    // TODO: execute remaining parts of leaf structure such as amountToReturn.
-    // TODO: emit events.
-
-    if relayer_refund_leaf.refund_accounts.len() != relayer_refund_leaf.refund_amounts.len() {
+    if relayer_refund_leaf.refund_addresses.len() != relayer_refund_leaf.refund_amounts.len() {
         return err!(CommonError::InvalidMerkleLeaf);
     }
 
-    // Derive the signer seeds for the state. The vault owns the state PDA so we need to derive this to create the
-    // signer seeds to execute the CPI transfer from the vault to the refund recipient.
-    let state_seed_bytes = state.seed.to_le_bytes();
-    let seeds = &[b"state", state_seed_bytes.as_ref(), &[ctx.bumps.state]];
-    let signer_seeds = &[&seeds[..]];
+    if ctx.remaining_accounts.len() < relayer_refund_leaf.refund_addresses.len() {
+        return err!(ErrorCode::AccountNotEnoughKeys);
+    }
 
-    // Will include in the emitted event at the end if there are any claim accounts.
-    let mut deferred_refunds = false;
+    // Check if vault has sufficient balance for all the refunds.
+    let total_refund_amount: u64 = relayer_refund_leaf.refund_amounts.iter().sum();
+    if ctx.accounts.vault.amount < total_refund_amount {
```

Comment:
Question: Do we want to consider the hub pool liabilities in the amount available to transfer?

Reply:
Was considering to add this amount initially, but that could still cause race to claim if vault balance decreased due to slow fills or claims from prior leaf executions. So the check here is more for the sake of consistency with EVM, but its effectiveness is bit limited.